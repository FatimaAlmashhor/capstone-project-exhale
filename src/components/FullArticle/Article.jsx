import React from 'react';
import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';
import { articles } from '../../services/fakeArticleService';

const Article = () => {
  const { t } = useTranslation();
  const { id } = useParams();

  const article = articles.find((p) => p.id === Number(id));
  let articleData;

  if (article) {
    articleData = (
      <div>
        <section className="blog ">
          <div className="container px-5 py-24 mx-auto">
            <div className="flex flex-wrap sm:-m-4 -mx-4 -mb-10 -mt-4">
              <div className="p-5  md:mb-0 mb-6 flex flex-col justify-center items-center  mx-auto">
                {article.articleImg && (
                  <div
                    className="bg-gray-300 h-56 w-full rounded-lg shadow-md bg-cover bg-center "
                    style={{ backgroundImage: `url(${article.articleImg}` }}
                  />
                )}
                <div className="  bg-white -mt-10 shadow-lg rounded-lg overflow-hidden p-5">
                  <div className="header-content inline-flex">
                    <div className="category-badge flex-1  h-4 w-4 m rounded-full m-1 bg-green-100">
                      <div className="h-2 w-2 rounded-full m-1 bg-green-500" />
                    </div>
                    <div className="category-title flex-1 text-sm text-gray-400">
                      {article.date}
                    </div>
                  </div>
                  <div className="title-post text-3xl">
                    <h2 className="text-blue-500 ">{article.title}</h2>
                  </div>
                  <div className="summary-post text-base text-justify text-1xl ">
                    <p>{article.text}</p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  } else {
    articleData = <h2>{t('Sorry. Article does not exist')}</h2>;
  }

  return (
    <div>
      <div>{articleData}</div>
    </div>
  );
};

export default Article;
