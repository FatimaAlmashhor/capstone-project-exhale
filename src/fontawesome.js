import { library } from '@fortawesome/fontawesome-svg-core';
import {
  faTwitter,
  faFacebook,
  faLinkedin,
  faInstagram,
  faGithub,
} from '@fortawesome/free-brands-svg-icons';

import { faCheck, faChevronLeft } from '@fortawesome/free-solid-svg-icons';

library.add(
  faTwitter,
  faFacebook,
  faLinkedin,
  faInstagram,
  faCheck,
  faChevronLeft,
  faGithub
);

// Import example:<FontAwesomeIcon icon={['fab', 'twitter']} />
