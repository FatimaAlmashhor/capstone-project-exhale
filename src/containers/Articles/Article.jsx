import React from 'react';
import FullArticle from '../../components/FullArticle/Article';

const Article = (artId) => {
  return (
    <div className="bg-background-secondary">
      <FullArticle id={artId} />
    </div>
  );
};

export default Article;
