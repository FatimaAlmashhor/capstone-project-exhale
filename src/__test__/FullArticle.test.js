import React from 'react';
import renderer from 'react-test-renderer';
import '../fontawesome';
import { BrowserRouter as Router } from 'react-router-dom';
import Article from '../components/FullArticle';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Article id={1} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
