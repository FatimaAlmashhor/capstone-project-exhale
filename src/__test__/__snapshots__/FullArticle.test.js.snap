// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <h2>
      Sorry. Article does not exist
    </h2>
  </div>
</div>
`;
